r/R/matrix.R

```diff
@@ -158,6 +157,76 @@ all_matrix_inputs <- function(x) {
   x
 }
 
+#' Get the max of each row in an iterable matrix
+#' @param x IterableMatrix object/dgCMatrix object
+#' @return *vector of maxes for every row
+#' @describeIn IterableMatrix-methods Calculate rowMax (replacement for `matrixStats::rowMax()`)
+#' @export
+rowMaxs <- function(x, rows = NULL, cols = NULL, na.rm = FALSE, ...) UseMethod("rowMaxs")
+#' @export
+rowMaxs.default <- function(x, rows = NULL, cols = NULL, na.rm = FALSE, ...) {
+  if (requireNamespace("MatrixGenerics", quietly = TRUE)) {
+    MatrixGenerics::rowMaxs(x, rows = rows, cols = cols, na.rm = na.rm, ...)
+  } else if (requireNamespace("matrixStats", quietly = TRUE)) {
+    matrixStats::rowMaxs(x, rows = rows, cols = cols, na.rm = na.rm, ...)
+  }
+  else {
+    stop("Can't run rowMaxs on a non-BPCells object unless MatrixGenerics or matrixStats are installed.")
+  }
+}
+#' @export
+rowMaxs.IterableMatrix <- function(x, rows = NULL, cols = NULL, na.rm = FALSE, ...) {
+  if(!is.null(rows) || !is.null(cols) || !isFALSE(na.rm)) {
+    stop("rowMaxs(IterableMatrix) doesn't support extra arguments rows, cols, or na.rm")
+  }
+  iter <- iterate_matrix(convert_matrix_type(x, "double"))
+  if(x@transpose == TRUE) {
+    res <- matrix_max_per_col_cpp(iter)
+  } else {
+    res <- matrix_max_per_row_cpp(iter)
+  }
+  names(res) <- rownames(x)
+  return(res)
+}
+if (requireNamespace("MatrixGenerics", quietly=TRUE)) {
```

Comment:
This should be wrapped in `rlang::on_load` (so it runs at every package load, not just at install time; this handles installing BPCells then MatrixGenerics later)

Reply:
Just realized I did that with one of the functions but not the other.. my bad!